astropy/table/table.py

```diff
@@ -3110,12 +3110,13 @@ def rename_columns(self, names, new_names):
 
     def _set_row(self, idx, colnames, vals):
         try:
-            assert len(vals) == len(colnames)
-        except Exception:
+            if len(vals) != len(colnames):
```

Comment:
@taldcroft do you prefer reverting to this revision or no `raise-from` at all, just the original message?

Reply:
I prefer keeping the original since this PR should be strictly confined to removing `assert` statements without any other changes.